### PR TITLE
KOS implementation

### PR DESCRIPTION
**Pull request by aliskin**
I added an implementation of KOS algorithm.

## Description
This commit includes implementation of KOS algorithm along with a few basic tests. Let me know if I need to make more tests.
The implementation was tested on the two datasets from [this paper](http://i.cs.hku.hk/~ydzheng2/crowd_truth_inference/crowd_truth_infer.pdf) and it achieves almost the same F1 and accuracy but works much faster.

I noticed that other aggregation algorithms have a parameter tol – should I add it to this method too?
Another question is about normalization. On each iteration the absolute values of intermediate grow and when there are many iterations the intermediate values might become nan. Any ideas on that point a very appreciated. Thanks! 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation and examples improvement (changes affected documentation and/or examples)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: crowdkit/aggregation/classification/kos.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    Attributes:
+        labels_ (Optional[pd.Series]): Tasks' labels.
+            A pandas.Series indexed by `task` such that `labels.loc[task]`
+            is the tasks's most likely true label.
+
+    """
+
+    n_iter: int = attr.ib(default=100)
+    random_state: int = attr.ib(default=0)
+
```
Comment (dustalov) [inline crowdkit/aggregation/classification/kos.py:58]: Please define all the attributes here, including the ones like `labels_` that we write in `fit()` and other methods. `MajorityVote` might be a good example.

File: tests/aggregation/test_kos_aggregation.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import pandas as pd
+import pytest
+from pandas.testing import assert_series_equal
+from crowdkit.aggregation import KOS
+
+
+@pytest.mark.parametrize(
+    'n_iter', [10, 100]
+)
+def test_aggregate_kos_on_data_with_bool_labels(n_iter, data_with_bool_labels, bool_labels_ground_truth):
```
Comment (dustalov) [inline tests/aggregation/test_kos_aggregation.py:15]: Please add type annotations everywhere.

File: crowdkit/aggregation/classification/kos.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                A pandas.DataFrame containing `task`, `worker` and `label` columns.
+        Returns:
+            KOS: self.
+        """
+
+        np.random.seed(self.random_state)
+
+        # Early exit
+        if not data.size:
+            self.labels_ = pd.Series(dtype=float)
```
[the remark this answers is not in the document]
Comment (dustalov) [reply crowdkit/aggregation/classification/kos.py:72]: It seems that we do the same thing in `DawidSkene`, but I believe that was a typo.

**Review by dustalov — CHANGES_REQUESTED** [review]

**Comment by dustalov** [comment]
And also please address the merge conflict.

File: crowdkit/aggregation/classification/kos.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                break
+            kos_data['reliabilities'] = (kos_data.summed - kos_data.multiplied).astype('float')
+
+        kos_data['inferred'] = kos_data.label * kos_data.reliabilities
+        inferred_labels = np.sign(kos_data.groupby('task')['inferred'].sum())
+        back_mapping = {v: k for k, v in mapping.items()}
+        self.labels_ = inferred_labels.apply(lambda x: back_mapping[x])
+        return self
+
+    def fit_predict(self, data: pd.DataFrame) -> pd.DataFrame:
```
[the remark this answers is not in the document]
Comment (aliskin) [reply crowdkit/aggregation/classification/kos.py:108]: Yes, I think we can think of them as logits. The meaning is that the bigger values come from more reliable workers according to our belief.

[the remark this answers is not in the document]
Comment (dustalov) [reply crowdkit/aggregation/classification/kos.py:108]: Does it affect the code?

[the remark this answers is not in the document]
Comment (pilot7747) [reply crowdkit/aggregation/classification/kos.py:108]: I don't see `fit_predict_proba` method, maybe we can just return softmax?

[the remark this answers is not in the document]
Comment (dustalov) [reply crowdkit/aggregation/classification/kos.py:108]: These numbers are not used as logits in the method. I propose not doing this now, however the change does not look difficult.

**Review by dustalov — APPROVED** [review]
